compiler/src/emit/invoke.rs

```diff
@@ -134,10 +134,18 @@ pub fn emit_function_invocation(
 
     let (env, captures) = match function_call.definition {
         Definition::User(id) => {
-            let captures = ctx.captures[id.0]
-                .as_ref()
-                .expect("captures not set during function invocation emission");
-            let env = ctx.engine.get_environment(id).unwrap();
+            let captures: &[ResolvedSymbol] = if function_call.reef != ctx.current_reef {
+                &[]
+            } else {
+                ctx.captures[id.0]
+                    .as_ref()
+                    .expect("captures not set during function invocation emission")
```

Comment:
```suggestion
                    .expect("undefined captures when the function is emitted")
```
*During* is for long time frames in English, and *emission* is a bit too French for me.